resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/ResourceApplication.java

```diff
@@ -16,24 +16,118 @@
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Application entrypoint and main process runner
+ */
 @EnableDiscoveryClient
 @SpringBootApplication
+@EnableConfigurationProperties({ProducerTopicConfiguration.class, ConsumerTopicConfiguration.class})
 public class ResourceApplication {
     private static final Logger LOGGER = LoggerFactory.getLogger(ResourceApplication.class);
 
-    public static void main(final String[] args) {
-        LOGGER.debug("ResourceApplication started with: {}", Arrays.toString(args));
+    private final Set<RunnableGraph<?>> runners;
+    private final Materializer materializer;
+    private final Executor executor;
+    private final PersistenceLayer persistence;
+    private final Supplier<List<Entry<Resource, LeafResource>>> resourceBuilder;
 
+    /**
+     * Autowire Akka objects in constructor for application ready event
+     *
+     * @param runners       collection of all Akka {@link RunnableGraph}s discovered for the application
+     * @param materializer  the Akka {@link Materializer} configured to be used
+     * @param persistence     a {@link PersistenceLayer} for persisting resources in, as if it were a cache
+     * @param executor      an executor for any {@link CompletableFuture}s (preferably the application task executor)
+     * @param resourceBuilder a {@link Supplier} of resources as built by a {@link uk.gov.gchq.palisade.service.ResourcePrepopulationFactory}, but with a connection detail attached
+     */
+    public ResourceApplication(final Collection<RunnableGraph<?>> runners,
+                               final Materializer materializer,
+                               final PersistenceLayer persistence,
+                               @Qualifier("configuredResourceBuilder") final Supplier<List<Entry<Resource, LeafResource>>> resourceBuilder,
+                               @Qualifier("applicationTaskExecutor") final Executor executor) {
+        this.runners = new HashSet<>(runners);
+        this.materializer = materializer;
+        this.persistence = persistence;
+        this.executor = executor;
+        this.resourceBuilder = resourceBuilder;
+    }
+
+    /**
+     * Application entrypoint, creates and runs a spring application, passing in the given command-line args
+     *
+     * @param args command-line arguments passed to the application
+     */
+    public static void main(final String[] args) {
+        LOGGER.debug("ResourceApplication started with: {}", (Object) args);
         new SpringApplicationBuilder(ResourceApplication.class).web(WebApplicationType.SERVLET)
                 .run(args);
     }
+
+    /**
+     * Adds resource(s) from a configuration file to the persistence of the {@link uk.gov.gchq.palisade.service.ResourceService}
+     */
+    @EventListener(ApplicationReadyEvent.class)
+    public void initPostConstruct() {
```

Comment:
Merge the 2 ApplicationReadyEvent classes, no point having 2, first prepopulation and then start kafka

Reply:
Both of these events have been merged into 1 method.